### ISSUE TITLE
Numerical column marked as Category breaks Number visualization formatting

### ISSUE THREAD

**Issue by attekei** [metabase/metabase#4566]
If field's Special Type is Category, then the formatting of Number visualization fails. In this example, I'm using the sample dataset and taking a sum of `Subtotal` field which is a Category by default (as discussed in #4565). The separators are not displayed as they should:

<img width="754" alt="screen shot 2017-03-21 at 13 27 51" src="https://cloud.githubusercontent.com/assets/6034928/24169358/97fcd60e-0e3a-11e7-9fbc-65e3e9f22394.png">

Only prefix and suffix settings work:

<img width="876" alt="screen shot 2017-03-21 at 13 34 51" src="https://cloud.githubusercontent.com/assets/6034928/24169544/3784c0ce-0e3b-11e7-9d0c-0879c2c80f63.png">

Removing the Category type from the column fixes the problem.

**Comment by tlrobinson** [comment]
@metabase/core-developers What's the correct fix here, not marking as categories, or forcing number formatting even if it's marked as a category.

Note that certain types such as zip code definitely should prevent formatting as a number (you want `90210` not `90,210`)

**Comment by mazameli** [comment]
In the general case, seems like we'd ideally want to solve this with more specific subtypes, like `ZIP code`, etc. But I think part of the specific problem that @attekei pointed out though was that types in the sample dataset were incorrect assigned: `Subtotal` in `Orders` should be `Number`, not `Category`.

**Comment by tlrobinson** [comment]
I believe this is due to the recent increase in the category cardinality cutoff (say that 3 times fast) to 300. The sample dataset has 199 unique subtotals.

I think the metadata sync heuristics need to be changed, but I'm not sure simply excluding numeric fields from categories is ideal.

**Comment by salsakran** [comment]
I don't think floats should ever be considered `category`s.

For ints, the only situation where we should treat them as `categories` is when they are really representing enums or similar situations.

I agree that this seems like a situation where we need subtypes.

**Comment by tlrobinson** [comment]
Makes sense. How can we detect enums? Maybe a much smaller cardinality threshold?

**Comment by cdeweyx** [comment]
Unable to repro this in v53 ([here](https://stats.metabase.com/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjoxLCJ0eXBlIjoicXVlcnkiLCJxdWVyeSI6eyJzb3VyY2UtdGFibGUiOiJjYXJkX18yNDI2OCJ9fSwiZGlzcGxheSI6InNjYWxhciIsImRpc3BsYXlJc0xvY2tlZCI6dHJ1ZSwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnsidGFibGUucGl2b3RfY29sdW1uIjoiSUQiLCJ0YWJsZS5jZWxsX2NvbHVtbiI6IlNVQlRPVEFMIiwic2NhbGFyLmZpZWxkIjoiUVVBTlRJVFkifSwidHlwZSI6InF1ZXN0aW9uIn0=)) — Going to close this for now, and if you are still able to repro please do leave a comment and we'll re-open 🙏
